src/utils.js

```diff
@@ -892,8 +892,9 @@ export function deepEqual(obj1, obj2, {checkTypes = false} = {}) {
     (typeof obj2 === 'object' && obj2 !== null) &&
     (!checkTypes || (obj1.constructor === obj2.constructor))
   ) {
-    if (Object.keys(obj1).length !== Object.keys(obj2).length) return false;
-    for (let prop in obj1) {
+    const props1 = Object.keys(obj1);
+    if (props1.length !== Object.keys(obj2).length) return false;
+    for (let prop of props1) {
```

Comment:
Take this as a suggestion (I don't know to to suggest changes for code outside the diff, so I made it a commit). Since we already run Object.keys I think this should save a couple cpu cycles.